java/src/main/java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -507,22 +507,20 @@ public static boolean isValidCode(String code) {
     if (separatorPosition != code.lastIndexOf(SEPARATOR)) {
       return false;
     }
-
-    if (separatorPosition % 2 != 0) {
+    // There must be an even number of at most 8 characters before the separator.
+    if (separatorPosition % 2 != 0 || separatorPosition > SEPARATOR_POSITION) {
       return false;
     }
 
```

Comment:
The changes below are minor code improvements to use the already defined SEPARATOR_POSITION constant, and to remove useless null checks since indexOf returns a primitive int (cannot be null) and invalid characters (when -1) are already validated after this.